apps/api/v2/src/modules/users/services/users.service.ts

```diff
@@ -19,7 +23,14 @@ export class UsersService {
       }
     }
 
-    return users;
+    // Validate each user response
+    usersFiltered.forEach((user) => {
+      if (!validateUserResponse(user)) {
+        throw new Error("Invalid user response schema");
```

Comment:
```suggestion
  throw new Error(`Invalid user response schema for user: ${user.username}`);
```

we can throw the error for the user.